### PR TITLE
Cpp 17

### PR DESCRIPTION
**Pull request by phunkyfish**
7.0.1
- Remove p8-platforms and add local StringUtils for cpp17

### CODE REVIEW

File: src/vbox/VBox.cpp [hunk 1]
```diff
@@ -284,7 +283,7 @@ void VBox::BackgroundUpdater()
       RetrieveGuide();
 
     lapCounter++;
-    usleep(5000 * 1000); // for some infinitely retarded reason, std::thread::sleep_for doesn't work
+    std::this_thread::sleep_for(std::chrono::milliseconds(5000));
```
Comment (Jalle19) [inline src/vbox/VBox.cpp:286]: What was the reason?

**Review by Jalle19 — COMMENTED** [review]
Have you checked whether we need StringUtils at all or if you could do without it? We probably only use one or two functions.

File: src/vbox/VBox.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/vbox/VBox.cpp:286]: I'm not sure, but I tested it and it's working. Maybe since C++11 it's working correctly.

Comment (AlwinEsch) [reply src/vbox/VBox.cpp:286]: There it has used the `usleep` before where was simulated by p8-platform for all OS. 
As there it is already possible by C++11 by this is it no more needed and can done with `std::this_thread::sleep_for`.

**Comment by phunkyfish** [comment]
It uses CompareNoCase and ToLower functions so it seemed reasonable to include it. Alternative is to just add these functions to utilities. If you want I can do that. But I went for quick and dirty.

**Comment by AlwinEsch** [comment]
Is not nice, but currently OK for me.
There comes also in next time a improvement and it can be out again. But as long the p8-platform bring problems and the addons are not creatable, should be a fix present. This here is the most easiest and if needed also to use other functions where maybe becomes needed.

**Comment by phunkyfish** [comment]
Stripping the used function out and putting them in Utilities is messy so I think this is the cleanest solution for now.

**Review by AlwinEsch — APPROVED** [review]

**Pull request merged**